### ISSUE TITLE
Add StringBuilder Span-based APIs

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#22430]
Separated out of https://github.com/dotnet/corefx/issues/21281 for tracking purposes.

- [x] Implement in System.Private.CoreLib in coreclr ("shared" with corert)
- [x] Expose from System.Runtime contract in corefx
- [x] Add tests to System.Runtime tests in corefx

```C#
namespace System.Text
{
    public class StringBuilder
    {
        public StringBuilder Append(ReadOnlySpan<char> value);
        public StringBuilder Insert(int index, ReadOnlySpan<char> value);
        public void CopyTo(int sourceIndex, Span<char> destination, int count);
        …
    }
}
```

**Comment by KooKiz** [comment]
Can we start working on these?

**Comment by stephentoub** [comment]
Yup.

**Comment by KooKiz** [comment]
Alright, on it.

**Comment by stephentoub** [comment]
Fixed by dotnet/coreclr#13163 and #22852
